### PR TITLE
Feature/recipe class

### LINKED ISSUE #2: Create Recipe Class
- It should hold on to all its information (provided in the data file).
- It should have methods to:
  - Determine the names of ingredients needed
  - Get the cost of its ingredients
  - Return its directions / instructions

### LINKED ISSUE #3: Add estimatedCostInCents method
To calculate the cost of a recipe, you will need to factor in the amount listed in the recipe. The price (estimatedCostInCents) of an ingredient is per unit.

For example, if flour is marked as 100 cents, and the recipe calls for 1.5 cups, then it would cost 150 cents. Similarly, if a tomato is $2, and the recipe requires 2 tomatoes, then the cost would be $4.

### PR DESCRIPTION
**Pull request by mattruder**
What's this PR do?
It adds methods to recipe class and associated test files

Where should the reviewer start?
Recipe.js and Recipe-test.js

How should this be manually tested?
Run npm test when in the main directory

Any background context you want to provide?
N/A

What are the relevant tickets?
resolve #2 
resolve #3 

Screenshots (if appropriate)
N/A

Questions:
None

### CODE REVIEW

File: test/Recipe-test.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  beforeEach(() => {
+    recipe = new Recipe(recipeData)
+    recipeRepository = new RecipeRepository(recipeData)
+  });
+
+  it('Should be a function', () => {
+    expect(Recipe).to.be.a('function');
+  });
+
+
```
Comment (aominhlong) [inline test/Recipe-test.js:56]: Remove extra spacing

File: test/RecipeRepository-test.js [hunk 2]
```diff
@@ -21,10 +23,10 @@ describe('Recipe', () => {
     recipeRepository.getRecipeByTag(`hor d'oeuvre`)
     expect(recipeRepository.recipes.length).to.equal(9)
   })
-  it('should filter recipe by name', () => {
+  it.skip('should filter recipe by name', () => {
 
     recipeRepository.getRecipeByName(`Dirty Steve`)
     expect(recipeRepository.recipes.length).to.equal(1)
     console.log(recipeRepository.recipes)
```
Comment (aominhlong) [inline test/RecipeRepository-test.js:30]: remove console.log

**Review by nhodnett — COMMENTED** [review]
Nice work Crew! Great that we got this functionality working without the need for an ingredients class. Will keep it in mind for future though and add if needed. That would resolve issue #4

File: src/classes/Recipe.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  let ingredientCosts = []
+  this.ingredients.map((recipeIngredient) => {
+    let filteredIngredients = this.ingredientsList.filter((ingredient) => {
+      if(ingredient.id === recipeIngredient.id) {
+        let cost = ingredient.estimatedCostInCents * recipeIngredient.quantity.amount;
+        ingredientCosts.push(cost)
+      }
+
+    })
+
```
Comment (aominhlong) [inline src/classes/Recipe.js:38]: Remove extra spacing

File: test/RecipeRepository-test.js [hunk 4]
```diff
@@ -2,11 +2,13 @@ import { expect } from 'chai';
 import RecipeRepository from '../src/classes/RecipeRepository';
 import recipeData from '../src/data/recipes.js';
 
-describe('Recipe', () => {
+describe('Recipe Repository', () => {
 
   let recipeRepository;
+  //let recipe;
```
Comment (aominhlong) [inline test/RecipeRepository-test.js:8]: Remove comments

**Pull request merged**